src/tests/ov_helpers/ov_snippets_models/src/subgraph_mha.cpp

```diff
@@ -69,12 +70,18 @@ std::shared_ptr<ov::Model> MHAFunction::initOriginal() const {
     const auto transpose1 = std::make_shared<ov::op::v1::Transpose>(transpose1Param, transpose1Const);
     std::shared_ptr<ov::Node> matmul_parent1 = transpose1;
     if (with_mul) {
-        ov::Shape shape(rank, 1);
-        if (transpose1->get_output_partial_shape(0).is_static()) {
-            shape[rank - 3] = transpose1->get_output_shape(0)[rank - 3];
+        if (is_mul_const) {
```

Comment:
Same as with the tests. It looks like the subgraph builder getting more and more complicated, so we should probably consider just creating a separate class. Otherwise, if the builders are getting complicated, we might create a bug inside the builder itself.  Ideally they should be very trivial to eliminate this possibility.

Reply:
Created new class